src/coordinates/uniform_cartesian.hpp

```diff
@@ -58,98 +58,77 @@ class UniformCartesian {
     cell_volume_ = dx_[0] * dx_[1] * dx_[2];
   }
 
-  template <class... Args>
-  KOKKOS_FORCEINLINE_FUNCTION Real Volume(Args... args) const {
-    return cell_volume_;
-  }
-
-  template <class... Args>
-  KOKKOS_FORCEINLINE_FUNCTION Real Dx(const int dir, Args... args) const {
+  template <int dir, class... Args>
+  KOKKOS_FORCEINLINE_FUNCTION Real Dxc(Args... args) const {
     assert(dir > 0 && dir < 4);
     return dx_[dir - 1];
   }
 
-  template <class... Args>
-  KOKKOS_FORCEINLINE_FUNCTION Real Area(const int dir, Args... args) const {
+  template <int dir, int face, class... Args>
+  KOKKOS_FORCEINLINE_FUNCTION Real Dxf(Args... args) const {
+    assert(dir > 0 && dir < 4 && face > 0 && face < 4);
+    return dx_[face - 1];
+  }
+
+  template <int dir>
+  KOKKOS_FORCEINLINE_FUNCTION Real Xc(const int idx) const {
     assert(dir > 0 && dir < 4);
-    return area_[dir - 1];
+    return xmin_[dir - 1] + (idx + 0.5) * dx_[dir - 1];
   }
```

Comment:
Let's add in those overloaded operators with all three coordinate directions. You shouldn't have to change Phoebus for lack of those operators.

Originally we thought about only using the full `(k,j,i)` operators, perhaps enforcing those so that those non-orthogonal coordinates would work in the future. I agree that those operators are also safer and less error-prone. However, there was push back so I decided to hold off doing that for now. Since Parthenon (and AthenaPK) only uses the `(i)` convention, I didn't bother adding `(k,j,i)` operators. Since you're asking, I'll add them.

Are `(i)` and `(k,j,i)` versions sufficient? I found the `<class... Args>` versions obtuse to work with but I can try.

Reply:
`i` and `(k,j,i)` should be sufficient. :+1: